pkg/gui/controllers/files_controller.go

```diff
@@ -748,6 +754,117 @@ func (self *FilesController) createStashMenu() error {
 	})
 }
 
+func (self *FilesController) openCopyMenu() error {
+	node := self.context().GetSelected()
+
+	copyNameItem := &types.MenuItem{
+		Label: self.c.Tr.CopyFileName,
+		OnPress: func() error {
+			name := node.Name()
+			return self.copyToClipboard(name)
+		},
+		Key: 'n',
+	}
+	copyPathItem := &types.MenuItem{
+		Label: self.c.Tr.CopyFilePath,
+		OnPress: func() error {
+			return self.copyToClipboard(node.Path)
+		},
+		Key: 'p',
+	}
+	copyFileDiffItem := &types.MenuItem{
+		Label:   self.c.Tr.CopySelectedDiff,
+		Tooltip: self.c.Tr.CopyFileDiffTooltip,
+		OnPress: func() error {
+			path, hasStaged := self.context().GetSelectedPath(), self.hasPathStagedChanges(node)
+			diff, err := self.c.Git().Diff.GetPathDiff(path, hasStaged)
+			if err != nil {
+				return self.c.Error(err)
+			}
+			return self.copyToClipboard(diff)
+		},
+		Key: 's',
+	}
+	copyAllDiff := &types.MenuItem{
+		Label:   self.c.Tr.CopyAllFilesDiff,
+		Tooltip: self.c.Tr.CopyFileDiffTooltip,
+		OnPress: func() error {
+			diff, err := self.c.Git().Diff.GetAllDiff(self.c.Helpers().WorkingTree.AnyStagedFiles())
+			if err != nil {
+				return self.c.Error(err)
+			}
+			return self.copyToClipboard(diff)
+		},
+		Key: 'a',
+	}
+
+	if node == nil {
+		copyNameItem.DisabledReason = self.c.Tr.FileCopyNoItemSelectedError
+		copyPathItem.DisabledReason = self.c.Tr.FileCopyNoItemSelectedError
+		copyFileDiffItem.DisabledReason = self.c.Tr.FileCopyNoItemSelectedError
+	}
+	if node != nil && !self.anyPathTrackedFile(node) {
+		copyFileDiffItem.DisabledReason = self.c.Tr.NoContentToCopyError
+	}
+	if !self.anyTrackedFile() {
+		copyAllDiff.DisabledReason = self.c.Tr.NoContentToCopyError
+	}
+
+	return self.c.Menu(types.CreateMenuOptions{
+		Title: self.c.Tr.CopyToClipboardMenu,
+		Items: []*types.MenuItem{
+			copyNameItem,
+			copyPathItem,
+			copyFileDiffItem,
+			copyAllDiff,
+		},
+	})
+}
+
+func (self *FilesController) anyPathTrackedFile(node *filetree.FileNode) bool {
+	stagedFiles := node.GetStagedFiles()
+	if len(stagedFiles) == 0 {
+		return node.SomeFile(func(t *models.File) bool {
+			return t.Tracked
+		})
+	}
+
+	for _, file := range stagedFiles {
+		if file.Tracked {
+			return true
+		}
+	}
+	return false
```

Comment:
This is not quite the logic we need here. For an untracked file, staging it is good enough to get a diff from it, so this block can just be replaced by `return true`. The function name is then not quite accurate any more, so maybe use `StagedOrTracked` instead of just `Tracked`, or maybe `StagedOrUnstagedChanges` (if we move it to FileNode, see above).

Also, you don't need `GetStagedFiles` any more, `GetHasStagedChanges` is enough.